### ISSUE TITLE
[FEA] Support try_multiply function

### ISSUE THREAD

**Issue by LIN-Yu-Ting** [NVIDIA/spark-rapids#11868]
I wish spark rapids can support try_multiply sql function:

```
try_multiply(CAST((GenotypeDT.alleleDepths[1]/GenotypeDT.depth) AS DECIMAL(5,4)), 100)
```

The fallback message is:

```
! <TryEval> tryeval(CheckOverflow((promote_precision(cast(cast((cast(alleleDepths#1912[1] as double) / cast(depth#1920 as double)) as decimal(5,4)) as decimal(7,4))) * 100.0000), DecimalType(9,4))) cannot run on GPU because GPU does not currently support the operator class org.apache.spark.sql.catalyst.expressions.TryEval
```

**Comment by revans2** [comment]
@LIN-Yu-Ting what version of Spark is this for? And do you need/want the exception handling to include the full expression tree?

`try_multiply` was added to Spark 3.3.0 https://issues.apache.org/jira/browse/SPARK-38164, but was then changed in Spark 3.4.0 https://issues.apache.org/jira/browse/SPARK-40222 so that numeric arguments would not catch exceptions for their children.

It looks like you are using a version of Spark 3.3.0 or after and before 3.4.0. If it is after 3.4.0 we have a bug in our code for versions 3.4.0 and above. I filed https://github.com/NVIDIA/spark-rapids/issues/11884 for it. Thanks for having me look more closely at the code.

To try and support TryEval generally, especially for 3.3.X to get try_multiply to work we are going to have to modify the entire tree to some how support this processing.

**Comment by LIN-Yu-Ting** [comment]
@revans2 you are correct that we are using Spark 3.3.0 in this case. Currently, we only apply try_multiply. Therefore, we do not need the exception handling to include the full expression tree. Thanks for your support.
